### ISSUE TITLE
Upcoming Neovim Job API Change

### ISSUE THREAD

**Issue by starcraftman** [junegunn/vim-plug#195]
Tarruda alerted us to change to neovim's api. Preview can be found tarruda/neovim@1af9d93. I'm going to look at it later and see what changes need doing.

Of important note from docs:
```
+jobstart({argv}[, {opts}]) {Nvim} *jobstart()*
+ Spawns {argv}(list) as a job. If passed, {opts} must be a
+ dictionary with the any of the following keys:
+ - on_stdout: stdout event handler
+ - on_stderr: stderr event handler
+ - on_exit: exit event handler
+ - user: user data passed to all callbacks
+ - pty: If set, the job will be connected to a new pseudo
+ terminal, and the job streams are connected to the master
+ file descriptor.
+ - width: Width of the terminal screen(only if pty is set)
+ - height: Height of the terminal screen(only if pty is set)
+ - TERM: $TERM environment variable(only if pty is set)
+ Either funcrefs or function names can be passed as event
+ handlers.
```

**Comment by starcraftman** [comment]
Working branch with changes required can be found at starcraftman/vim-plug@21f4245

**Comment by junegunn** [comment]
Cool, thanks!

@tarruda Is there a way to tell if the user is running an old (or an incompatible) version of neovim? I wish to warn the user and fall back to non-parallel installer in that case instead of throwing a bunch of errors.

**Comment by tarruda** [comment]
> @tarruda Is there a way to tell if the user is running an old (or an incompatible) version of neovim? I wish to warn the user and fall back to non-parallel installer in that case instead of throwing a bunch of errors.

I suppose you can use `exists('v:job_data')` to check if its old job API, but since Neovim is still alpha I think its reasonable to expect our users to constantly update to the master branch

**Comment by junegunn** [comment]
@tarruda JobActivity will go away, right? Then we can use `exists('##JobActivity')`.

**Comment by tarruda** [comment]
> @tarruda JobActivity will go away, right? Then we can use exists('##JobActivity')

:+1:

**Comment by junegunn** [comment]
@starcraftman So, I guess we should update the definition of `s:nvim`. By the way, simply unsetting `s:nvim` will activate Python installer on neovim. Last time I checked, it didn't work correctly, but I didn't look further into it.

**Comment by starcraftman** [comment]
@junegunn Changed s:nvim check, I believe that's all you seemed to imply. https://github.com/junegunn/vim-plug/compare/master...starcraftman:new_job_api#diff-0467dc3d01e54437b08868e627565c40R78

I just did a quick check of their python interface, doing `:py print 'hello'` fails. I think I'll check back with neovim later if I can't even get that working.

**Comment by junegunn** [comment]
@starcraftman 

> I believe that's all you seemed to imply.

That's right, thanks. And if we can't get python installer to work on neovim, we'll have to disable it as well; unsetting `s:py2`. But, AFAIK, python interface of neovim should not be any different from that of the original vim from the user's point of view, so it's probably something neovim guys should look into.

**Comment by starcraftman** [comment]
@junegunn Weird. The error I was getting with `py print 'hello'` has dissapeared. Instead I now get:
`Unhandled exception in thread started by <bound method PlugThread.__bootstrap of <PlugThread(PlugT-15, stopped 140
343022450432)>> `

I'll look into this, if I can't resolve by time of pulling this branch I'll just disable it.

**Comment by vheon** [comment]
I use more than one plugin written in python (YouCompleteMe, Ultisnips, ...) in neovim without problem, so it seems strange :S

**Comment by starcraftman** [comment]
@junegunn To summarize what I found. The neovim python if_pyth does not allow spawned threads to transparently access/modify vim.current.buffer (which is what we do in python/ruby) unlike in vim. See neovim/python-client#23 for my discussion. Since it isn't trivial, I'll disable for now. 

I may investigate later if simple changes to my Buffer class make it go.

**Comment by junegunn** [comment]
@starcraftman :+1:

**Comment by junegunn** [comment]
Closed by #198. Many thanks to @starcraftman.
